src/main/webapp/js/datahandlers/collegepagedatahandler.js

```diff
@@ -71,22 +72,24 @@ const loadCollegeData = async (element) => {
   try {
     json = await loadCollegeJson_();
   } catch(err) {
-    console.log(`Failed to load college data: ${err}`);
-    // TODO: Add a call to render an error message template to 
-    //   the DOM in place of the college template.
+    const occurrence = 'A network error has occurred. Failed to load college data.';
+    const action = 'Please reload the page or select a different college.';
+    console.log(`${occurrence} Error Message: ${err}.`);
+    renderErrorPage(element, occurrence, action, err);
   }
 
   try {
     const data = await convertFromJsonToTemplate_(json);
     const html = collegepage(data);
     element.innerHTML = html;
   } catch(err) {
-    console.log(`Failed to render college data: ${err} `)
-    // TODO: Add a call to render an error message template to 
-    //   the DOM in place of the college template.
+    const occurrence = 'A database error occurred. Failed to render college data.';
+    const action = 'The database failed to retrieve the college. \
+      This college may not exist. Please reload the page or select a different college.';
+    console.log(`${occurrence} Error Message: ${err}.`);
+    renderErrorPage(element, occurrence, action, err);
```

Comment:
Nice case handling. I'm just thinking if it will be better if we throw all errors to homepagecontroller.js and handle errors there. (Your way also work. We can save the design change to next week when the structure in other parts become clearer.)

Reply:
Yeah I agree. Handling errors in the homepagecontroller.js will be a good idea.